### PR TITLE
Add warning when attempting server connection without user permissions

### PR DESCRIPTION
**Pull request by strickvl**
Added a warning that catches the `IllegalOperationError` that we previously handed to the user without context/explanation.

Following discussion with @AlexejPenner, we agreed that there aren't really any scenarios we could imagine where a user might want to be logged into a ZenML server without any permissions attached to their role.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by schustmi** [comment]
Is this something that we can add a test for or is that unreasonably complicated?

**Comment by AlexejPenner** [comment]
> Is this something that we can add a test for or is that unreasonably complicated?

I don't think the test harness currently allows for attempting to connect to a new deployed zenml server. It also feels like unproportional effort to create this scenario for a niche exception like this one.

**Comment by schustmi** [comment]
What I had in mind, but might not be possible:
When the test are running with a rest zen server, create a new user without role. Then run CLI `connect` command to connect to that already connected server (or disconnect first if necessary) which should trigger this? But I now realize we only log a warning which isn't testable anyway I guess

**Comment by strickvl** [comment]
> What I had in mind, but might not be possible: When the test are running with a rest zen server, create a new user without role. Then run CLI `connect` command to connect to that already connected server (or disconnect first if necessary) which should trigger this? But I now realize we only log a warning which isn't testable anyway I guess

Technically we could test that the `IllegalOperationError` doesn't get raised...

**Comment by schustmi** [comment]
Probably not worth it though, maybe if it works after 5 mins it's a nice test but not that important to have I guess

**Comment by strickvl** [comment]
@AlexejPenner @schustmi added a test. Not sure if it'll work, but let's see.

File: src/zenml/cli/server.py [hunk 1]
```diff
@@ -785,7 +786,15 @@ def connect(
...
-    GlobalConfiguration().set_store(store_config)
+    try:
+        GlobalConfiguration().set_store(store_config)
+    except IllegalOperationError:
+        cli_utils.warning(
+            f"User '{username}' has no permissions attached to its role with "
+            f"which to access the server at '{url}'. Please ask the server "
+            f"administrator to assign a role with permissions to your "
+            f"username."
+        )
```
Comment (stefannica) [inline src/zenml/cli/server.py:797]:
```suggestion
    except IllegalOperationError as e:
        cli_utils.warning(
            f"User '{username}' does not have sufficient permissions to "
            f"to access the server at '{url}'. Please ask the server "
            f"administrator to assign a role with permissions to your "
            f"username: {str(e)}"
        )
```

Or something similar. The reason being that the IllegalOperationError can be thrown in cases where the user only has read permissions or doesn't have permissions to modify the current project, or other situations such as these, so the message should be more generic and also include the original error.

**Review by stefannica — CHANGES_REQUESTED** [review]

File: tests/integration/functional/cli/test_server.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    with does_not_raise():
+        result = runner.invoke(
+            cli.commands["connect"],
+            [
+                f"--url={server_url}",
+                f"--user={new_user_name}",
+                f"--password={new_password}",
+            ],
+        )
+        assert result.exit_code == 0
```
Comment (stefannica) [inline tests/integration/functional/cli/test_server.py:64]: Forgot to mention this the last time, but could you please delete the user at the end of the test, to not clutter the external servers with left-over users ?

**Review by stefannica — APPROVED** [review]

**Pull request merged**